internal/service/controller_service.go

```diff
@@ -332,66 +284,70 @@ func (s *ControllerService) Status(req *pb.StatusRequest, stream pb.ControllerSe
 
 	defer watcher.Stop()
 
-	for result := range watcher.ResultChan() {
-		switch result.Type {
-		case watch.Added, watch.Modified, watch.Deleted:
-			exporter = result.Object.(*jumpstarterdevv1alpha1.Exporter)
-			leased := exporter.Status.LeaseRef != nil
-			leaseName := (*string)(nil)
-			clientName := (*string)(nil)
-
-			original := client.MergeFrom(exporter.DeepCopy())
-			if false ||
-				meta.SetStatusCondition(&exporter.Status.Conditions, metav1.Condition{
-					Type:               string(jumpstarterdevv1alpha1.ExporterConditionTypeOnline),
-					Status:             metav1.ConditionTrue,
-					ObservedGeneration: exporter.Generation,
-					Reason:             "Connect",
-					Message:            "Exporter is connected to the Status endpoint",
-				}) ||
-				meta.SetStatusCondition(&exporter.Status.Conditions, metav1.Condition{
-					Type:               string(jumpstarterdevv1alpha1.ExporterConditionTypeRegistered),
-					Status:             metav1.ConditionTrue,
-					ObservedGeneration: exporter.Generation,
-					Reason:             "Connect",
-					Message:            "Exporter is connected to the Status endpoint",
-				}) {
-				if err = s.Client.Status().Patch(ctx, exporter, original); err != nil {
-					logger.Error(err, "unable to update exporter status")
+	ticker := time.NewTicker(time.Second * 10)
+
+	defer ticker.Stop()
+
+	online := func() {
+		original := client.MergeFrom(exporter.DeepCopy())
+		exporter.Status.LastSeen = metav1.Now()
+		if err = s.Client.Status().Patch(ctx, exporter, original); err != nil {
+			logger.Error(err, "unable to update exporter status.lastSeen")
+		}
+	}
+
+	select {
```

Comment:
What is the reason for this select before the other loop?

Can you explain with comments in code?. We need to improve our code comments, those don't need to be comprehensive, but some are useful.

Reply:
Ahh, to speed up marking it as online?

Reply:
Yes, otherwise it would take at most one minute before it's marked online.